packages/server-boilerplate/src/routes/Route.ts

```diff
@@ -41,12 +57,30 @@ export class Route<
     // swallowed.
     try {
       const response = await this.buildResponse();
-      this.respond(response, 200);
+      if (this.type === 'download') {
+        this.download(response);
+      } else {
+        this.respond(response, 200);
+      }
     } catch (error) {
       this.next(error);
     }
   }
 
+  private download = (response: DownloadResBody) => {
+    const { filePath = `download_${Date.now()}`, contents, type } = response;
+
+    if (type === '.json') {
+      writeJsonFile(filePath, contents);
+    } else {
+      fs.writeFileSync(filePath, contents);
+    }
+
+    this.res.download(filePath, () => {
+      fs.unlinkSync(filePath);
+    });
+  };
```

Comment:
@edmofro tagging this as candidate for refactoring after https://github.com/beyondessential/tupaia/pull/2965 is merged (can use `respondWithDownload` from that PR)

Reply:
Good call!